lib/index.js

```diff
@@ -11,6 +12,42 @@ var defaults = {
 
 exports.register = function(plugin, options, next) {
 
+  // handlers for reporting the different formats of incoming log messages
+  // this can be expanded to support an api-like interface via logging statements
+  var MessageHandlers = {
+    add : function(loggedMsg, done){
+      plugin.plugins.metrics.add(loggedMsg.data.name, loggedMsg.data.data, done);
+    },
+    remove : function(loggedMsg, done){
+      plugin.plugins.metrics.add(loggedMsg.data.name, loggedMsg.data.data, done);
+    }
+  };
+  var appropriateHandlerForMessageFormat = function(loggedMsg){
+    if (_.contains(loggedMsg.tags, 'add-metric')){
+      if (loggedMsg.data && loggedMsg.data.data && loggedMsg.data.name)
+        return MessageHandlers.add;
+    }
+    if (_.contains(loggedMsg.tags, 'remove-metric')){
+    }
+  };
+
+  var done = function(err,savedMsgObj){
+    if (err) plugin.log(['hapi-metrics', 'error'], {msg: err});
+  }
+
+  var handleLoggedMsg = function(loggedMsg){
+    var handler = appropriateHandlerForMessageFormat(loggedMsg);
+    var done = loggedMsg.data.callback ? loggedMsg.data.callback : done;
+    if (handler)
+      handler(loggedMsg, done);
+  };
+  var logEventHandler = function(loggedMsg){
+    if (!_.contains(loggedMsg.tags, 'metrics')) return;
+    handleLoggedMsg(loggedMsg);
+  }
+  // will be called every time a logging statement is made
+  plugin.on('log', logEventHandler);
```

Comment:
maybe we should put this code in a separate file.  Also, lets make the default not watch logs and they can enable by passing in options